### PR TITLE
Dark daily chart 

### PR DESCRIPTION
**Pull request by Emmo213**
Updated the daily chart to support only the light/dark themes based on the settings file.

### CODE REVIEW

**Comment by alexbelgium** [comment]
Works nicely! It looks slightly dull though... What do you think of making the species colors shades of red instead of shades of gray? As this might then recall the colors of the logos ?

image

**Comment by Emmo213** [comment]
Personally a "dark" theme to me is mostly black, white, and grey, so I don't think having reds and pinks would fit.

**Comment by Nachtzuster** [comment]
@alexbelgium , @Hiemstra87 does this match the rest of the dark 'colour' scheme, or are there some tweaks needed?

**Comment by alexbelgium** [comment]
I was trying to make the background slightly darker (darkgrey or dimgrey) as I thought it was a bit luminous but I can't find which piece of the code to modify ?

**Comment by Emmo213** [comment]
Sure, how does this look?

Current:
image

Proposed:
image

**Comment by alexbelgium** [comment]
I very much like the proposed one - I think it highlights in a nice fashion the birds graphs. It is easier in my opinion to read it
It is also a color that seems very aligned with the rest of the theme

Thanks for checking that !

**Comment by Emmo213** [comment]
Changed pushed. I actually used the "darkgrey" color that you suggested. With the darker background I was also able to make the current hour a different color again. 

image

**Comment by alexbelgium** [comment]
> @alexbelgium , @Hiemstra87 does this match the rest of the dark 'colour' scheme, or are there some tweaks needed?

Ready to merge for me! Thanks @Emmo213

**Comment by Emmo213** [comment]
Not sure what the linter is unhappy about. Let me install it locally to verify.

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v3
         with:
-          python-version: '3.9.x'
+          python-version: '3.11.x'
```
Comment (Nachtzuster) [inline .github/workflows/python-ci.yml:20]: This needs to stay on 3.9: people that migrated their installation in place from the original BirdNET-Pi are still on 3.9.

File: scripts/daily_plot.py [hunk 2]
```diff
@@ -43,7 +45,11 @@ def show_values_on_bars(ax, label):
         # Species Count Total
         value = '{:n}'.format(p.get_width())
         bbox = {'facecolor': 'lightgrey', 'edgecolor': 'none', 'pad': 1.0}
-        ax.text(x, y, value, bbox=bbox, ha='center', va='center', size=9, color='darkgreen')
+        match conf['COLOR_SCHEME']:
```
Comment (Nachtzuster) [inline scripts/daily_plot.py:48]: Match was only introduced with Python 3.10; this was giving the syntax error earlier. Could you change to a if/else?

File: scripts/update_birdnet_snippets.sh [hunk 3]
```diff
@@ -180,6 +180,7 @@ fi
 if [ -L /usr/local/bin/analyze.py ];then
   rm -f /usr/local/bin/analyze.py
 fi
+
```
Comment (Nachtzuster) [inline scripts/update_birdnet_snippets.sh:183]: Please don't do changes in unrelated files

File: scripts/daily_plot.py [hunk 4]
```diff
@@ -70,9 +76,15 @@ def create_plot(df_plt_today, now, is_top=None):
...
     df_plt_selection_today = df_plt_today[df_plt_today.Com_Name.isin(plt_selection_today.index)]
 
+    conf = get_settings()
+
     # Set up plot axes and titles
     height = max(readings / 3, 0) + 1.06
-    f, axs = plt.subplots(1, 2, figsize=(10, height), gridspec_kw=dict(width_ratios=[3, 6]), facecolor='#77C487')
+    match conf['COLOR_SCHEME']:
+        case "dark":
+            f, axs = plt.subplots(1, 2, figsize=(10, height), gridspec_kw=dict(width_ratios=[3, 6]), facecolor='darkgrey')
```
Comment (Nachtzuster) [inline scripts/daily_plot.py:85]: Could you move `f, axs = ...` out the case and only assign facecolor?

File: scripts/daily_plot.py [hunk 5]
```diff
@@ -102,7 +119,7 @@ def create_plot(df_plt_today, now, is_top=None):
 
     # Generate frequency plot
     plot = sns.countplot(y='Com_Name', hue='Com_Name', legend=False, data=df_plt_selection_today,
-                         palette=colors, order=freq_order, ax=axs[0])
+                         palette=colors, order=freq_order, ax=axs[0], edgecolor='lightgrey')
```
Comment (Nachtzuster) [inline scripts/daily_plot.py:122]: the edge color is set for the default theme here too, I guess this is not the intent here?

**Review by Nachtzuster — CHANGES_REQUESTED** [review]
Finally got some time to look into this, thanks for your patience!

File: .github/workflows/python-ci.yml (hunk 1 above)
Comment (Emmo213) [reply .github/workflows/python-ci.yml:20]: 3.9 is about 4 years old. When can we move to a more up to date version? Should updating Python and/or other dependencies be part of the update script.

File: scripts/daily_plot.py (hunk 2 above)
Comment (Emmo213) [reply scripts/daily_plot.py:48]: Correct, which is why I moved the version to 3.11. 

If we're not updating the Python version I can change this to an if/else - it's just a less elegant solution.

File: scripts/daily_plot.py (hunk 5 above)
Comment (Emmo213) [reply scripts/daily_plot.py:122]: That was intentional. While it is more pronounced with the dark theme even with the green theme there are times the bar from the bar chart is white or almost white and blends into the chart background. Adding the border around the bar helps it stand out from the background.

image

File: scripts/update_birdnet_snippets.sh (hunk 3 above)
Comment (Emmo213) [reply scripts/update_birdnet_snippets.sh:183]: When there is general code cleanup do you want that in a separate PR? In this case it's adding a blank line between two functions that didn't have the blank line and should have. 

image

File: .github/workflows/python-ci.yml (hunk 1 above)
Comment (alexbelgium) [reply .github/workflows/python-ci.yml:20]: What Nachtzuster means (I think) is that birdnet pi has an incremental auto update feature. So if we change the dependency to a newer version of python incremental update won't work and it will break functionality for older users

Comment (Emmo213) [reply .github/workflows/python-ci.yml:20]: I just think at some point we're going to want to have an update, if not for better Python features than to avoid any potential security issues. It doesn't need to be bleeding edge but generally speaking staying back leveled isn't ideal.

**Comment by Emmo213** [comment]
Back leveled to Python 3.9.

File: scripts/daily_plot.py (hunk 5 above)
Comment (Nachtzuster) [reply scripts/daily_plot.py:119]: yes, it does look better with the edge

File: scripts/daily_plot.py [hunk 6]
```diff
@@ -43,7 +45,10 @@ def show_values_on_bars(ax, label):
         # Species Count Total
         value = '{:n}'.format(p.get_width())
         bbox = {'facecolor': 'lightgrey', 'edgecolor': 'none', 'pad': 1.0}
-        ax.text(x, y, value, bbox=bbox, ha='center', va='center', size=9, color='darkgreen')
+        if conf['COLOR_SCHEME'] == "dark":
+            ax.text(x, y, value, bbox=bbox, ha='center', va='center', size=9, color='black')
+        else:
+            ax.text(x, y, value, bbox=bbox, ha='center', va='center', size=9, color='darkgreen')
```
Comment (Nachtzuster) [inline scripts/daily_plot.py:51]:
```suggestion
        if conf['COLOR_SCHEME'] == "dark":
            color='black'
        else:
            color='darkgreen'
        ax.text(x, y, value, bbox=bbox, ha='center', va='center', size=9, color=color)
```

File: scripts/daily_plot.py [hunk 7]
```diff
@@ -70,9 +75,14 @@ def create_plot(df_plt_today, now, is_top=None):
...
 
+    conf = get_settings()
+
     # Set up plot axes and titles
     height = max(readings / 3, 0) + 1.06
-    f, axs = plt.subplots(1, 2, figsize=(10, height), gridspec_kw=dict(width_ratios=[3, 6]), facecolor='#77C487')
+    if conf['COLOR_SCHEME'] == "dark":
+        f, axs = plt.subplots(1, 2, figsize=(10, height), gridspec_kw=dict(width_ratios=[3, 6]), facecolor='darkgrey')
+    else:
+        f, axs = plt.subplots(1, 2, figsize=(10, height), gridspec_kw=dict(width_ratios=[3, 6]), facecolor='none')
```
Comment (Nachtzuster) [inline scripts/daily_plot.py:85]:
```suggestion
    if conf['COLOR_SCHEME'] == "dark":
        facecolor='darkgrey'
    else:
        facecolor='none'
    f, axs = plt.subplots(1, 2, figsize=(10, height), gridspec_kw=dict(width_ratios=[3, 6]), facecolor='darkgrey')
```

File: scripts/daily_plot.py (hunk 4 above)
Comment (Emmo213) [reply scripts/daily_plot.py]: Done.

File: scripts/daily_plot.py (hunk 7 above)
Comment (Emmo213) [reply scripts/daily_plot.py]: Change made using variable instead of hardcoded color
